src/metabase/sync/sync_metadata/tables.clj

```diff
@@ -140,27 +142,47 @@
   (log/info "Marking tables as inactive:"
             (for [table old-tables]
               (sync-util/name-for-logging (mi/instance Table table))))
-  (doseq [{schema :schema, table-name :name, :as _table} old-tables]
+  (doseq [{schema :schema table-name :name :as _table} old-tables]
     (t2/update! Table {:db_id  (u/the-id database)
                        :schema schema
                        :name   table-name
                        :active true}
                 {:active false})))
 
-(mu/defn ^:private update-table-description!
-  "Update description for any `changed-tables` belonging to `database`."
-  [database       :- i/DatabaseInstance
-   changed-tables :- [:set i/DatabaseMetadataTable]]
-  (log/info "Updating description for tables:"
-            (for [table changed-tables]
-              (sync-util/name-for-logging (mi/instance Table table))))
-  (doseq [{schema :schema, table-name :name, description :description} changed-tables]
-    (when-not (str/blank? description)
-      (t2/update! Table {:db_id       (u/the-id database)
-                         :schema      schema
-                         :name        table-name
-                         :description nil}
-                  {:description description}))))
+(mu/defn ^:private update-table-metadata-if-needed!
+  "Update the table metadata if it has changed."
+  [table-metadata :- i/DatabaseMetadataTable
+   metabase-table :- (ms/InstanceOf :model/Table)]
+  (log/infof "Updating table metadata for %s" (sync-util/name-for-logging metabase-table))
+  (let [to-update-keys [:description :database_require_filter]
+        old-table      (select-keys metabase-table to-update-keys)
+        new-table      (select-keys (merge
+                                     (zipmap to-update-keys (repeat nil))
+                                     table-metadata)
+                                    to-update-keys)
+        [_ changes _]  (data/diff old-table new-table)
+        changes        (cond-> changes
+                         ;; we only update the description if and only if the initial state is nil
+                         ;; because don't want to override the user edited description if it exists
+                         (some? (:description old-table))
```

Comment:
Before, the check was `(str/blank? ...)` not `(some? ...)`. Why did you change it?

Reply:
I think that was a mistake. If a table does not have a description, then the jdbc will return `NULL`.

whereas if a table has a description and it's empty, then it might be a case where the user manually removes the description in Metabase.

In this case, we don't want to override that on update.

Reply:
Makes sense!